x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/filestructurefinder/FileStructure.java

```diff
@@ -652,8 +680,14 @@ public FileStructure build() {
                     throw new IllegalStateException("enum value [" + format + "] missing from switch.");
             }
 
-            if ((timestampField == null) != (timestampFormats == null || timestampFormats.isEmpty())) {
-                throw new IllegalArgumentException("Timestamp field and timestamp formats must both be specified or neither be specified.");
+            if ((timestampField == null) != (jodaTimestampFormats == null || jodaTimestampFormats.isEmpty())) {
```

Comment:
nit: No need to change but I think this would read more easily if it was:

```if (timestampField != null && (jodaTimestampFormats == null || jodaTimestampFormats.isEmpty())```

Reply:
To be equivalent it would have to say:

```
if ((timestampField != null && (jodaTimestampFormats == null || jodaTimestampFormats.isEmpty())) || (timestampField == null && jodaTimestampFormats != null && jodaTimestampFormats.isEmpty() == false))
```

I could introduce some local variables and make it:

```
if (wasTimestampFieldSpecified != wasJodaTimestampFormatsSpecified)
```

Or I could split it over two `if` statements, which would allow for different error messages saying exactly which setting was specified and which wasn't.  (The reason I didn't bother with separate error messages originally is that this validation is not end-user validation - it's just early bug finding for developers who change the functionality in the future. But if the current code is too hard to read then I can.)

What do you think?

Reply:
I think the local variables approach would definitely make it clear enough.